pkg/messagebroker/kafka.go

```diff
@@ -639,3 +639,29 @@ func (k *KafkaBroker) AddTopicPartitions(ctx context.Context, request AddTopicPa
 		Response: resp,
 	}, nil
 }
+
+// IsHealthy checks the health of the kafka
+func (k *KafkaBroker) IsHealthy(ctx context.Context) (bool, error) {
+	doneCh := make(chan healthResponse)
+
+	go func() {
```

Comment:
can we create a context with Deadline of 1 second and pass that on to this method, if library handles the timeout defined based on ctx, we can avoid this

Reply:
Had thought of this. But the comment on top of the function impl warns the timeout may not be honored. Hence did it this way. Maybe I'm interpreting it wrong.

```
// Note on cancellation: Although the underlying C function respects the
// timeout, it currently cannot be manually cancelled. That means manually
// cancelling the context will block until the C function call returns.
```

https://github.com/confluentinc/confluent-kafka-go/blob/13ae1158cd0e098a1093c52d77d556a5c733e242/kafka/adminapi.go#L430

Reply:
i think it will handle the timeout value but we can't cancel context at our end, so we can use context.WithDeadline but we can't use context.WithCancel

Reply:
Made the change